shell/common/switches.h

```diff
@@ -176,6 +176,11 @@ DEF_SWITCH(UseTestFonts,
            "will make font resolution default to the Ahem test font on all "
            "platforms (See https://www.w3.org/Style/CSS/Test/Fonts/Ahem/). "
            "This option is only available on the desktop test shells.")
+DEF_SWITCH(DisableAssetFonts,
+           "disable-asset-fonts",
+           "Prevents usage of any non-test fonts unless they were explicitly "
+           "Loaded via dart:ui font APIs. This option is only available on the "
```

Comment:
How is it kept only available in the test shell?

Reply:
oh right. hmm. I think should probably ifdef this out

Reply:
well, I guess the thinking is that these switches aren't on an allowlist or surfaced through the embedder, but this one and use-test-fonts are technically still there. We should fix that

Reply:
ahh, but if we change a define for switches then we'd have to compile this twice for each host platform

🤔

Reply:
Added a debug mode guard